src/components/App/index.js

```diff
@@ -30,20 +29,22 @@ import 'semantic-ui-css/semantic.min.css';
 
 import Page from './Page';
 import NotFound from './NotFound';
+import trackedComponentEnahncer from './tracked-component-enhancer';
 import { ConnectedDependencyGraphPage } from '../DependencyGraph';
 import { ConnectedSearchTracePage } from '../SearchTracePage';
 import { ConnectedTracePage } from '../TracePage';
 import JaegerAPI, { DEFAULT_API_ROOT } from '../../api/jaeger';
 import configureStore from '../../utils/configure-store';
-import metricConfig from '../../utils/metrics';
 import prefixUrl from '../../utils/prefix-url';
 
 import './App.css';
 
-const PageWithMetrics = metrics(metricConfig)(Page);
-
 const defaultHistory = createHistory();
 
+const TrackedSearchPage = trackedComponentEnahncer(ConnectedSearchTracePage);
```

Comment:
Why break this up? If we do it at the top level container, we don't have to add it to each page.

If our concern is that the Page only mounts once, I would hook into React router page change event and do it there.

Reply:
Scratch my comment, didn't realize we were on react router v4.

Reply:
I'm wondering if we can do something similar to this helper library 
https://github.com/pshrmn/rrc/blob/master/docs/OnUpdate.md

It would be nice if we could listen to all changes on the location without explicitly putting it on each page component. That way, when a new page is added, it will automatically start tracking it without having to use this HOC. Thoughts?

Reply:
Hm, could possibly connect it to the store and use `state.location` instead... I'll look into that.

Reply:
Moved tracking into `Page`.